magenta-lib/src/main/scala/magenta/tasks/Task.scala

```diff
@@ -1,10 +1,12 @@
 package magenta
 package tasks
 
+import software.amazon.awssdk.services.sts.StsClient
+
 trait Task {
   // execute this task (should throw on failure)
-  def execute(reporter: DeployReporter, stopFlag: => Boolean)
-  def execute(reporter: DeployReporter) { execute(reporter, stopFlag = false) }
+  def execute(resources: DeploymentResources, stopFlag: => Boolean)
+  def execute(resources: DeploymentResources) { execute(resources, stopFlag = false) }
```

Comment:
Small side note - the second line here is a legacy interface and could be removed from the `Task` definition and the `stopFlag = false` added to the calls from tests. Not really part of this PR though so this comment is just noise 😏